src/doc/man/qtel.1
.TH QTEL "1" "AUGUST 2021" Linux "User Manuals"
.
.SH NAME
.
qtel \- a "QT EchoLink" client
.
.SH DESCRIPTION
.
Qtel stands for the "QT EchoLink" client. It is only a EchoLink client application.
There is no "sysop mode". If it is a link you want to run, have a look at the SvxLink
server application. Qtel is quite a simple application, so most things should be self
explainatory. There are three windows: the main window, the communications dialog
and the settings dialog.
.
.PP
.B The Qtel Main Window
.PP
The main window consists of four areas. The top-left is where to choose which class of
staions to view (Conferences, Links, Repeaters and private Stations). All stations in
a class is shown to the top-right. At left-bottom is a message area that shows the messages
from the EchoLink directory server. At right-bottom is a list of incoming connections. To
accept an incoming connection, highlight the station and press "Accept".
.
.PP
.B The Qtel Communications Dialog
.PP
The communications dialog is used to perform a Qso to anoher station. To activate it,
double-click on a station line or just press the enter key when the station is highlighted.
To connect to the station, press the Connect button. To transmit, press the PTT button.
To disconnect, press the "Disconnect" button. In the big, white area in the middle of the
window, info messages from the remote station are shown. Chat messages from the
remote station is also shown here. In the oneline white area below, chat messages can be
typed that is sent to the remote station.
.
.PP
.B The Qtel Settings Dialog
.PP
The configuration dialog is brought up the first time the application is started.
Most fields are self explainatory. Location is the short string that is shown in
the EchoLink directory server listing. The info message is the one that is sent to the
remote station upon connection. In the Directory Server tab are some settings for the EchoLink directory
server connection.
.
.SH OPTIONS
There are no options. Start the program within a console - qtel & - or from the menu.
.
.SH "SEE ALSO"
.nh
.ad l
For detailed description look at
.B https://github.com/sm0svx/svxlink/wiki/QtelUserDocs
.
.SH ENVIRONMENT
.
.TP
ASYNC_AUDIO_NOTRIGGER
Set this environment variable to 1 if you get an error about
.B ioctl: Broken pipe
during qtel startup when using OSS audio.
.TP
ASYNC_AUDIO_ALSA_ZEROFILL
Set this environment variable to 0 to stop the Alsa audio code from writing
zeros to the audio device when there is no audio to write available.
ASYNC_AUDIO_UDP_ZEROFILL
Set this environment variable to 1 to enable the UDP audio code to write zeros
to the UDP connection when there is no audio to write available.
.
.SH AUTHOR
.
Tobias Blomberg (SM0SVX) <sm0svx at svxlink dot org>
.PP
.nh
.ad l
This manual page was written by Michael Vorleiter
<parsival1609@yahoo.de>
.
.SH REPORTING BUGS
.
Bugs should be reported using the issue tracker at
https://github.com/sm0svx/svxlink.

Questions about SvxLink should not be asked using the issue tracker. Instead
use the group set up for this purpose at groups.io:
https://groups.io/g/svxlink


src/qtel/qtel.cpp
/**
@file	 qtel.cpp
@brief   An EchoLink client implemented in Qt
@author  Tobias Blomberg
@date	 2003-03-09

This is an EchoLink client implemented in Qt using the classes from EchoLib to
create a fully working EchoLink client.

\verbatim
Qtel - The Qt EchoLink client
Copyright (C) 2003-2009 Tobias Blomberg

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
\endverbatim
*/



/****************************************************************************
 *
 * System Includes
 *
 ****************************************************************************/

#include <cstdio>
#include <cstdlib>

#include <sigc++/sigc++.h>

#include <QApplication>
#include <QMessageBox>
#include <QTranslator>
#include <QTextCodec>
#include <QLocale>


/****************************************************************************
 *
 * Project Includes
 *
 ****************************************************************************/

#include <config.h>

#include <AsyncQtApplication.h>
#include <AsyncTcpClient.h>

#include <EchoLinkDirectory.h>
#include <EchoLinkDispatcher.h>

	
/****************************************************************************
 *
 * Local Includes
 *
 ****************************************************************************/

#include "Settings.h"
#include "MainWindow.h"


/****************************************************************************
 *
 * Namespaces to use
 *
 ****************************************************************************/

using namespace Async;
using namespace EchoLink;


/****************************************************************************
 *
 * Defines & typedefs
 *
 ****************************************************************************/

#define PROGRAM_NAME "qtel"

	

/****************************************************************************
 *
 * Prototypes
 *
 ****************************************************************************/



/****************************************************************************
 *
 * Global Variables
 *
 ****************************************************************************/
 


/****************************************************************************
 *
 * Private Global Variables
 *
 ****************************************************************************/



/****************************************************************************
 *
 * MAIN
 *
 ****************************************************************************/

/*
 *----------------------------------------------------------------------------
 * Function:  main
 * Purpose:   Start everything...
 * Input:     argc  - The number of arguments passed to this program
 *    	      	      (including the program name).
 *    	      argv  - The arguments passed to this program. argv[0] is the
 *    	      	      program name.
 * Output:    Return 0 on success, else non-zero.
 * Author:    Tobias Blomberg
 * Created:   2003-03-09
 * Remarks:   
 * Bugs:      
 *----------------------------------------------------------------------------
 */
int main(int argc, char **argv)
{
  QtApplication app(argc, argv);
  
  QTranslator translator;
  translator.load(QString("qtel_%1").arg(QLocale::system().name()),
      SHARE_INSTALL_PREFIX "/qtel/translations");
  app.installTranslator(&translator);

  QCoreApplication::setOrganizationName("SvxLink");
  QCoreApplication::setOrganizationDomain("open-link.org");
  QCoreApplication::setApplicationName("Qtel");
	
  Settings *settings = Settings::instance();
  bool cfg_ok = false;
  while (!cfg_ok)
  {
    settings->readSettings();
    if (settings->callsign().isEmpty() || settings->password().isEmpty() ||
	settings->name().isEmpty())
    {
      int button = QMessageBox::critical(0, "Bad configuration",
	  "There are one or more configuration items that have not been "
	  "filled in. Please fill in the missing items.",
	  "Configure", "Quit");
      if (button == 1)
      {
      	return 1;
      }
    }
    else
    {
      cfg_ok = true;
    }
  }
   
  MainWindow mainWindow;
  mainWindow.show();
  app.exec();
  
  delete Settings::instance();
  
  return 0;
}



/****************************************************************************
 *
 * Functions
 *
 ****************************************************************************/



/*
 * This file has not been truncated
 */
